### PR TITLE
Added options to retain info and state mqtt messages

### PR DESCRIPTION
**Pull request by jeroenst**
## Description:

With this pr it is possible to retain info and state messages, configurable with inforetain and stateretain commands in the console.

https://github.com/arendst/Tasmota/discussions/11067

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works on Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works on Tasmota core ESP32 V.1.0.5-rc6
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/settings.h [hunk 1]
```diff
@@ -146,8 +146,8 @@ typedef union {                            // Restricted by MISRA-C Rule 18.4 bu
     uint32_t zb_received_as_subtopic : 1;  // bit 4 (v9.2.0.3)   - SetOption118 - (Zigbee) Move ZbReceived from JSON message and into the subtopic replacing "SENSOR" default
     uint32_t zb_omit_json_addr : 1;        // bit 5 (v9.2.0.3)   - SetOption119 - (Zigbee) Remove the device addr from json payload, can be used with zb_topic_fname where the addr is already known from the topic
     uint32_t zb_topic_endpoint : 1;        // bit 6 (v9.2.0.4)   - SetOption120 - (Zigbee) Append endpoint number to topic if device dependent (use with SetOption89)
-    uint32_t spare07 : 1;                  // bit 7
-    uint32_t spare08 : 1;                  // bit 8
+    uint32_t mqtt_state_retain : 1;        // bit 7 (development)- CMND_STATERETAIN
```
Comment (s-hadinger) [inline tasmota/settings.h:149]: Please follow the same format is above and take the current version

**Review by s-hadinger — COMMENTED** [review]
Make sure to honor SetOption 104 no_retain and never send a retained message if it's set. Otherwise it will close the Mqtt connection

File: tasmota/settings.h (hunk 1 above)
Comment (jeroenst) [reply tasmota/settings.h:149]: is current version 9.3.0.1 ok to write here?

**Review by jeroenst — COMMENTED** [review]
> Make sure to honor SetOption 104 no_retain and never send a retained message if it's set. Otherwise it will close the Mqtt connection

This is honoration is already in the current code: https://github.com/arendst/Tasmota/blob/73e2e4ff9ca777d9dda5031a95cf0f5992323ebf/tasmota/xdrv_02_mqtt.ino#L326-L328

File: tasmota/settings.h (hunk 1 above)
Comment (s-hadinger) [reply tasmota/settings.h:149]: Yes

Comment (jeroenst) [reply tasmota/settings.h]: Ok, I also found it in tasmota_version.h,  thanx.

https://github.com/arendst/Tasmota/blob/73e2e4ff9ca777d9dda5031a95cf0f5992323ebf/tasmota/tasmota_version.h#L23

**Pull request closed without merging**